src/lakefs_spec/client_helpers.py

```diff
@@ -63,3 +65,21 @@ def revert(client: LakeFSClient, repository: str, branch: str, parent_number: in
     client.branches_api.revert_branch(
         repository=repository, branch=branch, revert_creation=revert_creation
     )
+
+
+def rev_parse(
+    client: LakeFSClient,
+    repository: str,
+    ref: str,
+    parent: int = 0,
+    prefixes: List[str] = [],
+    objects: List[str] = [],
+) -> str:
+    if parent < 0:
+        raise ValueError(f"Parent cannot be negative. Got {parent}")
+    revisions = client.refs_api.log_commits(
+        repository=repository, ref=ref, limit=2 * (parent + 1), prefixes=prefixes, objects=objects
+    ).results
+    if not revisions:
+        raise ApiException("Could not fetch commit log from lakeFS client.")
```

Comment:
Seems that it just 404s on an unknown ref, so we might just do this:

```suggestion
    try:
        revisions = client.refs_api.log_commits(
            repository=repository, ref=ref, limit=2 * (parent + 1), prefixes=prefixes, objects=objects
        ).results
    except NotFoundException:
        raise RuntimeError("{ref!r} does not match any revision in lakeFS repository {repository!r}")
```
Could also be a `ValueError`.

(You might have to format the code, I typed this blind in the comment field)